src/Http/ResponseBuilder.php

```diff
@@ -137,12 +169,27 @@ public function jsonSerialize()
      * @param  array $data
      * @return array
      */
+    protected function includeSuccessFlag(array $data):array
+    {
+        if (! $this->includeSuccessFlag) {
+            return $data;
+        }
+
+        return array_merge([], ['success' => $this->successFlag], $data);
+    }
+
+    /**
+     * Include a status code to the serialized data if enabled.
+     *
+     * @param  array $data
+     * @return array
+     */
     protected function includeStatusCode(array $data):array
     {
         if (! $this->includeStatusCode) {
             return $data;
         }
 
-        return array_merge(['status' => $this->statusCode], $data);
+        return array_merge([], ['status' => $this->statusCode], $data);
```

Comment:
Awe, it is a holder over from other attemps I had made. We can squash this and the other array_merge on `includeSuccessFlag`